libraries/botbuilder-dialogs-adaptive/src/recognizers/regexRecognizer.ts

```diff
@@ -94,81 +91,67 @@ export class RegexRecognizer extends Recognizer implements RegexRecognizerConfig
                 }
             }
 
-            if (matches.length > 0) {
-                const intentKey = intentPattern.intent.split(' ').join('_');
-                if (!recognizerResult.intents.hasOwnProperty(intentKey)) {
-                    recognizerResult.intents[intentKey] = { score: 1.0 };
-                    recognizerResult.intents[intentKey]['pattern'] = intentPattern.pattern;
-                }
+            if (matches.length === 0) {
+                continue;
+            }
 
-                for (let j = 0; j < matches.length; j++) {
-                    const match = matches[j];
-                    if (match.groups) {
-                        for (const name in match.groups) {
-                            const text = match.groups[name];
-                            if (text) {
-                                const entity: Entity = {
-                                    type: name,
-                                };
-                                entity['text'] = text;
-                                entity['start'] = match.index;
-                                entity['end'] = match.index + text.length;
-                                entityPool.push(entity);
-                            }
-                        }
-                    }
+            // TODO length weighted match and multiple intents
+            const intentKey = intentPattern.intent.split(' ').join('_');
+            (recognizerResult.intents[`${intentKey}`] as { score: number; pattern: string }) ??= {
```

Comment:
Type checking preferred to `as` casting.

Reply:
I ported `IntentScore` interface from C#, whose can be expanded to store properties other than `score: number`:
```
interface IntentScore {
  score?: number;
  [key: string]: unknown;
}
```
Changed the `intents` field of `RecognizerResult` from `{ [name: string]: { score: number } }` to `Record<string, IntentScore>` to reflect the change, I believe this would not break the backward compatibility.